.github/workflows/pin-requirements.yml

```
@@ -0,0 +1,48 @@
+name: Pin Requirements
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  update-pinned-requirements:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          ref: main
+
+      - uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435
+        with:
+          python-version: "3.x"
```

Per comment about minimum Python:

```suggestion
          # NOTE: Update whenever sigstore-python's minimum Python is bumped.
          python-version: "3.7"
```

---

This is currently set here, which ensures anytime we manually update this, we use the right version: https://github.com/sigstore/sigstore-python/blob/main/install/.python-version

I think we need to use something higher than 3.7 to capture all possible dependencies.

---

It looks like we can install with anything in 3.7 - 3.10; the big constraint is that we can't install 3.11, because we need the `importlib_resources` shim if the installing version is below 3.11:

https://github.com/sigstore/sigstore-python/blob/37483bcbc2b009f58d36103c368533dfe0fc5082/pyproject.toml#L32